File: Maths/subarray_sum_k.cxx
#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    static int subarraySum(vector<int> &arr, int k){
        int count = 0;
        for(auto i = 0; arr[i]; ++i){
            if(arr[i]==k)
                count++;
            int sum = arr[i];
            for(auto j = i+1; arr[j]; ++j){
                sum = sum+arr[j];
                if(sum==k)
                    count++;
            }
        }
        return count;
    }
};
int main(){
    ios_base::sync_with_stdio(0);
    cout.tie(0);
    vector<int> arr = {1,1,1};
    int k = 2;
    cout << Solution::subarraySum(arr,k);
}

File: Maths/subarray_sum_k.py
class Solution(object):
    def __init__(self,nums,k:int):
        self.nums = nums
        self.k = k
    def subarraySum(self,nums,k:int)->int:
        self.count = 0
        self.n = len(self.nums)
        for i in range(self.n):
            if(self.nums[i]==self.k):
                self.count+=1
            self.sum = self.nums[i]
            for j in range(i+1,self.n):
                self.sum = self.sum+self.nums[j]
                if(self.sum==self.k):
                    self.count+=1
        return self.count


if __name__=='__main__':
    test = [1,1,1]
    k = 2
    s = Solution(test,k)
    print(s.subarraySum(test,k)) # 2
